x/ibc/handler.go

```diff
@@ -37,22 +38,75 @@ func NewHandler(k Keeper) sdk.Handler {
 
 		// IBC channel msgs
 		case channel.MsgChannelOpenInit:
-			res, cap, err := channel.HandleMsgChannelOpenInit(ctx, k.ChannelKeeper, msg)
+			cbs := k.PortKeeper.LookupModule(ctx, msg.PortID)
+			portCap, _ := cbs.GetCapability(ctx, ibctypes.PortPath(msg.PortID))
+			err := cbs.OnChanOpenInit(ctx, msg.Channel.Ordering, msg.Channel.ConnectionHops, msg.PortID, msg.ChannelID, msg.Channel.Counterparty, msg.Channel.Version)
+			if err != nil {
+				return nil, err
+			}
+			res, cap, err := channel.HandleMsgChannelOpenInit(ctx, k.ChannelKeeper, portCap, msg)
+			cbs.ClaimCapability(ctx, cap, ibctypes.ChannelCapabilityPath(msg.PortID, msg.ChannelID))
```

Comment:
Module claims created channel capability using callback

Reply:
Instead of doing it this way, how about passing the channel capability to `OnChanOpenInit` ? I think that's cleaner.